core/pkg/resourcehandler/k8sresources.go

```diff
@@ -133,25 +143,62 @@ func (k8sHandler *K8sResourceHandler) GetResources(sessionObj *cautils.OPASessio
-			logger.L().Warning("failed to collect cloud data", helpers.Error(err))
+			logger.L().Warning("failed to collect cloud resources", helpers.Error(err))
 		}
-		if provider != "" {
-			if sessionObj.Metadata != nil && sessionObj.Metadata.ContextMetadata.ClusterContextMetadata != nil {
-				sessionObj.Metadata.ContextMetadata.ClusterContextMetadata.CloudProvider = provider
+	}
+
+	return k8sResourcesMap, allResources, ksResourceMap, nil
+}
+
+func (k8sHandler *K8sResourceHandler) collectCloudResources(sessionObj *cautils.OPASessionObj, allResources map[string]workloadinterface.IMetadata, ksResourceMap *cautils.KSResources, cloudResources []string) error {
+	var err error
+	clusterName := cautils.ClusterName
+	provider := cloudsupport.GetCloudProvider(clusterName)
+	if provider == "" {
+		return fmt.Errorf("failed to get cloud provider, cluster: %s", clusterName)
+	}
+	if sessionObj.Metadata != nil && sessionObj.Metadata.ContextMetadata.ClusterContextMetadata != nil {
+		sessionObj.Metadata.ContextMetadata.ClusterContextMetadata.CloudProvider = provider
+	}
+	logger.L().Debug("cloud", helpers.String("cluster", clusterName), helpers.String("clusterName", clusterName), helpers.String("provider", provider))
+
+	for resourceKind, resourceGetter := range cloudResourceGetterMapping {
+		if cloudResourceRequired(cloudResources, resourceKind) {
+			logger.L().Debug("Collecting cloud data ", helpers.String("resourceKind", resourceKind))
+			wl, err := resourceGetter(clusterName, provider)
+			if err != nil {
+				if !strings.Contains(err.Error(), cloudv1.NotSupportedMsg) {
+					// Return error with useful info on how to configure credentials for getting cloud provider info
+					logger.L().Debug("failed to get cloud data", helpers.String("resourceKind", resourceKind), helpers.Error(err))
+					err = fmt.Errorf("failed to get %s descriptive information. Read more: https://hub.armosec.io/docs/kubescape-integration-with-cloud-providers", strings.ToUpper(provider))
+					cautils.SetInfoMapForResources(err.Error(), cloudResources, sessionObj.InfoMap)
+				}
+			} else {
+				allResources[wl.GetID()] = wl
+				(*ksResourceMap)[fmt.Sprintf("%s/%s", wl.GetApiVersion(), wl.GetKind())] = []string{wl.GetID()}
 			}
 		}
+	}
 
-		// api server info resource
+	// get api server info resource
+	if cloudResourceRequired(cloudResources, string(cloudsupport.TypeApiServerInfo)) {
 		err = k8sHandler.collectAPIServerInfoResource(allResources, ksResourceMap)
 		if err != nil {
-			logger.L().Warning("failed to collect api server info resource", helpers.Error(err))
+			logger.L().Debug("failed to collect api server info resource", helpers.Error(err))
 		}
 	}
+	return err
+}
 
-	return k8sResourcesMap, allResources, ksResourceMap, nil
+func cloudResourceRequired(cloudResources []string, resource string) bool {
+	for _, cresource := range cloudResources {
+		if strings.Contains(cresource, resource) {
```

Comment:
[Question]: Are we sure we want this case-sensitive?

Reply:
I can change it. In K8s resources we are case-insensitive? I'll follow what we do currently

Reply:
We're actually currently [filtering cloud resources case-sensitively,](https://github.com/kubescape/kubescape/blob/e4e3071f5fd9f2f47f78de6f4b927b0310fe0606/core/cautils/workloadmappingutils.go#L34) (same with host and buln resources) so I'll leave it like this for now